girder/api/describe.py

```diff
@@ -377,3 +494,190 @@ def routeHandler(...)
+                        kwargs.pop(name, None)  # Remove from path params
+                        kwargs[info['destName']] = self._loadModel(name, info, params[name])
+                    else:
+                        kwargs[name] = self._validateParam(name, descParam, params[name])
+                    kwargs['params'].pop(name, None)  # Remove from form/query params
+                elif descParam['in'] == 'body':
+                    if name in self.description.jsonParams:
+                        info = self.description.jsonParams[name].copy()
+                        info['required'] = descParam['required']
+                        kwargs[name] = self._loadJsonBody(name, info)
+                    else:
+                        kwargs[name] = cherrypy.request.body
+                elif 'default' in descParam:
+                    kwargs[name] = descParam['default']
+                elif descParam['required']:
+                    raise RestException('Parameter "%s" is required.' % name)
+                else:
+                    # If required=False but no default is specified, use None
+                    if name in self.description.modelParams:
+                        info = self.description.modelParams[name]
+                        kwargs.pop(name, None)  # Remove from path params
+                        kwargs[info['destName']] = None
+                    else:
+                        kwargs[name] = None
+
+            if self.description.hasPagingParams and 'sort' in kwargs:
+                kwargs['sort'] = [(kwargs['sort'], kwargs['sortdir'])]
+                del kwargs['sortdir']
+
+            return fun(*args, **kwargs)
+
+        if self.hide:
+            wrapped.description = None
+        else:
+            wrapped.description = self.description
+        return wrapped
+
+    def _validateJsonType(self, name, info, val):
+        if info['requireObject'] and not isinstance(val, dict):
+            raise RestException('Parameter %s must be a JSON object.' % name)
+        if info['requireArray'] and not isinstance(val, list):
+            raise RestException('Parameter %s must be a JSON array.' % name)
+
+    def _loadJsonBody(self, name, info):
+        val = None
+        if cherrypy.request.body.length == 0 and info['required']:
+            raise RestException('JSON parameter %s must be passed in request body.' % name)
+        elif cherrypy.request.body.length > 0:
+            val = getBodyJson()
+            self._validateJsonType(name, info, val)
+
+        return val
+
+    def _loadJson(self, name, info, value):
+        try:
+            val = bson.json_util.loads(value)
+        except ValueError:
+            raise RestException('Parameter %s must be valid JSON.' % name)
+
+        self._validateJsonType(name, info, val)
+
+        return val
+
+    def _loadModel(self, name, info, id):
+        if id is None:
+            if info['required']:
```

Comment:
I thought I did that in 7a2a815, but that isn't `None`, it's actually the string `'None'`. I'm not sure it's possible to get this value to be `None` through the HTTP API, so I will remove this check altogether.